iothub/device/src/ModuleClient.cs

```diff
@@ -96,11 +101,12 @@ public static ModuleClient Create(string hostname, string gatewayHostname, IAuth
         /// <param name="hostname">The fully-qualified DNS hostname of IoT Hub</param>
         /// <param name="authenticationMethod">The authentication method that is used</param>
         /// <param name="transportSettings">Prioritized list of transportTypes and their settings</param>
+        /// <param name="options">The options that allow configuration of the module client instance during initialization.</param>
         /// <returns>ModuleClient</returns>
         public static ModuleClient Create(string hostname, IAuthenticationMethod authenticationMethod,
-            ITransportSettings[] transportSettings)
+            ITransportSettings[] transportSettings, ClientOptions options = default)
         {
```

Comment:
Just want to make sure that this defaults to the same code path as for device client i.e. modelId sits on connect packet for MQTT ? Have you verified it with sample or other ways?

Reply:
Also is it possible to e2e this?

Reply:
Yes this is using the same ClientFactory that is used for DeviceClient. So it's following the same code path. 
@prmathur-microsoft @abhipsaMisra Does the backend already support us sending modelId on device modules?

Reply:
Azure IoT Explorer would be the easiest way to test this. If it doesn't work, I'd reach out to Miguel Nunez Avila (from the pnp service team).

Reply:
Got it. I will write a small e2e test for module client and device client.

Reply:
As Binal hinted offline during a sync we were having, I cannot at the moment do an e2e test because testing happens against a specific canary region. So what I did is wrote an e2e test locally, and I was able to verify from IoT Explorer the following:
"dt-dataschema": "dtmi:my:company:namespace;1"

So once this feature becomes available in all regions, we can add an e2e test. I will create a task to track this.